sdk/storage/azure-storage-common/src/crypt.cpp

```diff
@@ -151,6 +151,10 @@ namespace Azure { namespace Storage {
 
     std::vector<uint8_t> Sha256(const std::vector<uint8_t>& data)
     {
+      if (data.size() > std::numeric_limits<ULONG>::max())
```

Comment:
How do we decide what's too big? Should this be a precondition instead of throwing, since it is easy for the caller to validate the input argument?

cc @JeffreyRichter

Reply:
Yes, a precondition seems right instead of throwing.

Reply:
https://github.com/Azure/azure-sdk-for-cpp/pull/2421